back/engines/commercial/multi_tenancy/lib/tasks/core/migrate_flexible_input_forms.rake

```diff
@@ -0,0 +1,170 @@
+    else
+      fields = merge_ideation_fields fields, custom_form
+    end
+
+    # Validate that sections / pages are present
+    errors = {}
+    IdeaCustomFieldsService.new(custom_form).check_form_structure fields, errors
+    if errors.present?
+      error_handler "Form structure does not validate #{errors}"
+      return
+    end
+
+    # Validate that the ordering is correct
+    ordering = fields.pluck('ordering')
+    if ordering[0] != 0 || !ordering.each_cons(2).all? { |x, y| y == x + 1 }
+      error_handler "Ordering is not correct - #{ordering}"
+      return
+    end
+
+    # Save the changes
+    fields.each do |field|
+      field_id = field.code || field.key
+      field_order = field.ordering
+      if field.persisted?
+        if field.changed?
+          Rails.logger.info "FIELD EXISTS - UPDATING: #{field_id} #{field.changes.keys}"
+          if persist_changes
+            if field.save
+              field.insert_at(field_order)
+            else
+              error_handler "Cannot update field - #{field_id} - #{field.id} - #{field.errors.errors}"
+            end
+          end
+          @stats[:updated] += 1
+        else
+          Rails.logger.info "FIELD EXISTS - NO CHANGES: #{field_id}"
+        end
+      else
+        Rails.logger.info "NEW FIELD - CREATING: #{field_id}"
+        if persist_changes
+          if field.save
+            field.insert_at(field_order)
+          else
+            error_handler "Cannot create field - #{field_id} - #{field.id} - #{field.errors.errors}"
+          end
+        end
+        @stats[:created] += 1
+      end
+    end
+  end
+
+  def error_handler(error)
+    Rails.logger.error "ERROR: #{error}"
+    @stats[:errors] << error
+  end
+
+  def merge_ideation_fields(fields, custom_form)
+    participation_method = Factory.instance.participation_method_for custom_form.participation_context
+    default_fields = participation_method.default_fields(custom_form)
+    constraints = participation_method.constraints
+    updated_fields = []
+
+    # Merge existing fields with same code as defaults
+    default_fields.each do |default_field|
+      existing_field = fields.find { |field| field.code == default_field.code }
+      updated_fields << if existing_field
+        merge_attributes(existing_field, default_field, constraints)
+      else
+        default_field
+      end
+    end
+
+    # Add any additional custom fields that don't match the defaults
+    fields.each do |field|
+      unless updated_fields.find { |f| f.id == field.id }
+        updated_fields << field
+      end
+    end
+    fix_ordering(updated_fields)
+
+    updated_fields
+  end
+
+  # Fix ordering in the context of a form, not of the platform
+  # Done here to trigger changed attribute, but acts_as_list.insert_at used to set later
+  def fix_ordering(fields)
+    order_before = fields.pluck('ordering')
+    fields.each_with_index do |field, index|
```

Comment:
I thought acts_as_list had a helper method for this, but I can't find it back right away.